components/mdns/mdns.c

```diff
@@ -6299,7 +6304,32 @@ esp_err_t mdns_service_txt_item_remove(const char *service, const char *proto, c
     return mdns_service_txt_item_remove_for_host(NULL, service, proto, NULL, key);
 }
 
-esp_err_t mdns_service_subtype_add_for_host(const char *instance_name, const char *service, const char *proto,
+static esp_err_t _mdns_service_subtype_remove_for_host(mdns_srv_item_t *service, const char *subtype)
+{
+    esp_err_t ret = ESP_ERR_NOT_FOUND;
+    mdns_subtype_t *srv_subtype = service->service->subtype;
+    mdns_subtype_t *pre = service->service->subtype;
+    while (srv_subtype) {
+        if (strcmp(srv_subtype->subtype, subtype) == 0) {
+            // Target subtype is found.
+            if (srv_subtype == service->service->subtype) {
+                // The first node needs to be removed
+                service->service->subtype = service->service->subtype->next;
+            } else {
+                pre->next = srv_subtype->next;
+            }
+            free((char *)srv_subtype->subtype);
+            free(srv_subtype);
+            ret = ESP_OK;
+        }
+        pre = srv_subtype;
+        srv_subtype = srv_subtype->next;
```

Comment:
Here might be a bug when remove some entries in the middle of the list, will fix it...

Reply:
Missing a `break`, added.